### PR TITLE
 Document alternative deploy method of Kubeflow Pipelines

### PR DESCRIPTION
**Pull request by Bobgy**
@jessiezcc @IronPan can you help me review/find someone to help me review this?
The new page can be found at https://deploy-preview-1174--competent-brattain-de2d6d.netlify.com/docs/pipelines/deployment-alternatives/

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/website/1174)

### CODE REVIEW

**Comment by Bobgy** [comment]
This is part of https://github.com/kubeflow/pipelines/issues/1638

File: content/docs/pipelines/pipelines-deploy-alternatives.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
++++
+title = "Pipelines Deploy Alternatives"
+description = "Alternative approaches to deploy just Kubeflow Pipelines to a cluster"
```
Comment (jingzhang36) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: maybe change "just"  to lightweight?

Comment (Bobgy) [reply content/docs/pipelines/pipelines-deploy-alternatives.md]: thanks, changed

File: content/docs/pipelines/pipelines-deploy-alternatives.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
++++
+title = "Pipelines Deploy Alternatives"
+description = "Alternative approaches to deploy lightweight Kubeflow Pipelines to a cluster"
```
Comment (sarahmaddox) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: s/deploy/deploying

"deploy" is a verb not a noun.

File: content/docs/pipelines/pipelines-deploy-alternatives.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+weight = 15
++++
+
+Despite [deploying Kubeflow](/docs/started/getting-started/#installing-kubeflow) as a
```
Comment (sarahmaddox) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: s/Despite/As an alternative to

File: content/docs/pipelines/pipelines-deploy-alternatives.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+whole with many components including pipelines, you also have a choice to deploy
+Kubeflow Pipelines' lite deployment. Pipelines has a kustomize manifests for
```
Comment (sarahmaddox) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: Are we planning to use the word "lite" as a brand for this type of deployment? That's a product management / marketing decision. I recommend that we remove all mention of "lite" and use "lightweight" instead every time.

Comment (sarahmaddox) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: Remove the apostrophe after "Pipelines'".

Comment (sarahmaddox) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: Suggestion for improved clarity: Change this:
"Pipelines has a kustomize manifests for this. You can follow the instructions below to deploy it in an existing Kubernetes cluster."

To this:
"Follow the instructions below to deploy Kubeflow Pipelines in an existing cluster, using the supplied kustomize manifest."

File: content/docs/pipelines/pipelines-deploy-alternatives.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+this. You can follow the instructions below to deploy it in an existing cluster.
+
+Knowledge about [Kubernetes](https://kubernetes.io/docs/home/), [kubectl](https://kubernetes.io/docs/reference/kubectl/overview/) and [kustomize](https://kustomize.io/) will help you understand this
+document better and be able to customize based on your needs.
+
+
+## TL;DR
```
Comment (sarahmaddox) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: The heading "TL;DR" here is confusing in two ways. Firstly, it implies that this section is a summary of what follows - but that doesn't seem right, as the first section deploys to an existing cluster whereas the later section involves creating a cluster. Secondly, not everyone knows what "TL;DR" means.

Suggested heading for better clarity:
"Deploying Kubeflow Pipelines to an existing cluster"

File: content/docs/pipelines/pipelines-deploy-alternatives.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+To deploy Kubeflow Pipelines in an existing Kubernetes cluster, do the following:
+
+1. Deploy latest version of Kubeflow Pipelines
+```
+export PIPELINE_VERSION={{% kfp-latest-version %}}
+kubectl apply -f https://storage.googleapis.com/ml-pipeline/pipeline-lite/$PIPELINE_VERSION/namespaced-install.yaml
+```
+
+2. Get the Pipeline UI URL
```
Comment (sarahmaddox) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: "Get the URL for the Kubeflow Pipelines UI"

File: content/docs/pipelines/pipelines-deploy-alternatives.md [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+Knowledge about [Kubernetes](https://kubernetes.io/docs/home/), [kubectl](https://kubernetes.io/docs/reference/kubectl/overview/) and [kustomize](https://kustomize.io/) will help you understand this
+document better and be able to customize based on your needs.
+
+
+## TL;DR
+
+To deploy Kubeflow Pipelines in an existing Kubernetes cluster, do the following:
+
+1. Deploy latest version of Kubeflow Pipelines
```
Comment (sarahmaddox) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: For readability, add a colon ":" at the end of this line and every similar line below.

File: content/docs/pipelines/pipelines-deploy-alternatives.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+    You have many options from cloud service providers. For GCP, you can reference
+    [Creating a cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/creating-a-cluster).
+
+    You can also set up your own Kubernetes cluster.
+
+    Cluster minimum requirement: node pool with at least 2 nodes and 2 CPUs per node.
+
+1. Download kubectl CLI tool
+
+    You can get kubectl from [its official doc](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
```
Comment (sarahmaddox) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: For readability, add a full stop "." at the end of this sentence and any similar sentences below.

File: content/docs/pipelines/pipelines-deploy-alternatives.md [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+2. Get the Pipeline UI URL
+```
+kubectl describe configmap inverse-proxy-config -n kubeflow | grep googleusercontent.com 
+```
+
+## Step by step tutorial to deploy Pipelines from scratch
+
+1. Prepare a Kubernetes cluster
+
+    You have many options from cloud service providers. For GCP, you can reference
```
Comment (sarahmaddox) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: Expand any abbreviation on first usage. So here:
s/GCP/Google Cloud Platform (GCP)

File: content/docs/pipelines/pipelines-deploy-alternatives.md [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+
+1. Download kubectl CLI tool
+
+    You can get kubectl from [its official doc](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+
+1. Configure kubectl to talk to your cluster
+
+    For GCP, you can reference [Configuring cluster access for kubectl](https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-access-for-kubectl)
+
+1. Deploy latest version of Kubeflow Pipelines Lite to your cluster
```
Comment (sarahmaddox) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: The capital letter in "Lite" implies that we have a branding decision to call this deployment "Kubeflow Pipelines Lite". Is that so? If yes, then use this branding throughout the doc. If not, then use "lightweight deployment" throughout.

File: content/docs/pipelines/pipelines-deploy-alternatives.md [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+
+    ```
+    export PIPELINE_VERSION={{% kfp-latest-version %}}
+    kubectl apply -k github.com/kubeflow/pipelines//manifests/kustomize/env/dev?ref=$PIPELINE_VERSION
+    ```
+
+    Kubeflow pipelines applications take a while (~3 minutes) to start.
```
Comment (sarahmaddox) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: "Kubeflow Pipelines" is a brand name, so use capital letters.

File: content/docs/pipelines/pipelines-deploy-alternatives.md [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+## Customization
+Customization can be done through Kustomize [Overlay](https://github.com/kubernetes-sigs/kustomize/blob/master/docs/glossary.md#overlay). 
+
+Note - The instruction below assume you installed kubectl v1.14.0 or later, which has native support of kustomize.
+To get latest kubectl, visit [here](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+
+For the following instructions, first clone [Kubeflow Pipelines repo](https://github.com/kubeflow/pipelines),
+and use it as working directory.
+
+### Deploy on GCP with CloudSQL and GCS
```
Comment (sarahmaddox) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: s/GCS/Google Cloud Storage
(According to GCP branding guidelines)

File: content/docs/pipelines/pipelines-deploy-alternatives.md [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+This is recommended for production environments. See
+[here](https://github.com/kubeflow/pipelines/tree/master/manifests/kustomize/env/gcp) for more details. 
+
+### Change deploy namespace
```
Comment (sarahmaddox) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: This heading implies the reader should change and deploy a namespace. 
s/deploy/deployment

File: content/docs/pipelines/pipelines-deploy-alternatives.md [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+To deploy Kubeflow Pipelines in namespace FOO,
+- Edit [dev/kustomization.yaml](https://github.com/kubeflow/pipelines/blob/master/manifests/kustomize/env/dev/kustomization.yaml)
```
Comment (sarahmaddox) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: This list does not render properly. See 
https://5d8c8937399d410009e3c0a4--competent-brattain-de2d6d.netlify.com/docs/pipelines/pipelines-deploy-alternatives/

File: content/docs/pipelines/pipelines-deploy-alternatives.md [hunk 15]
```diff
@@ -0,0 +1,139 @@
++++
+title = "Pipelines Deploy Alternatives"
```
[the remark this answers is not in the document]
Comment (sarahmaddox) [reply content/docs/pipelines/pipelines-deploy-alternatives.md]: Recommendation: Change the file name too. For example, change `/pipelines/pipelines-deploy-alternatives` to `/pipelines/deployment-alternatives`.

File: content/docs/pipelines/pipelines-deploy-alternatives.md [hunk 16]
```diff
@@ -0,0 +1,139 @@
...
+1. Deploy latest version of Kubeflow Pipelines Lite to your cluster
+
+    ```
+    export PIPELINE_VERSION={{% kfp-latest-version %}}
+    kubectl apply -k github.com/kubeflow/pipelines//manifests/kustomize/env/dev?ref=$PIPELINE_VERSION
+    ```
+
+    Kubeflow pipelines applications take a while (~3 minutes) to start.
+
+1. Get public URL of Pipelines UI and use it to use Kubeflow Pipelines
```
Comment (sarahmaddox) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: Suggested wording:
"Get the public URL of the Kubeflow Pipelines UI and use it to access Kubeflow Pipelines:"

File: content/docs/pipelines/pipelines-deploy-alternatives.md [hunk 17]
```diff
@@ -0,0 +1,139 @@
...
+    ```
+    kubectl describe configmap inverse-proxy-config -n kubeflow | grep googleusercontent.com 
+    ```
+
+
+## Customization
+Customization can be done through Kustomize [Overlay](https://github.com/kubernetes-sigs/kustomize/blob/master/docs/glossary.md#overlay).
```
Comment (sarahmaddox) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: s/Kustomize/kustomize

(The brand name is lower case.)

Comment (sarahmaddox) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: s/Overlay/overlays

File: content/docs/pipelines/pipelines-deploy-alternatives.md [hunk 18]
```diff
@@ -0,0 +1,139 @@
...
+- Edit [dev/kustomization.yaml](https://github.com/kubeflow/pipelines/blob/master/manifests/kustomize/env/dev/kustomization.yaml)
+or [gcp/kustomization.yaml](https://github.com/kubeflow/pipelines/blob/master/manifests/kustomize/env/gcp/kustomization.yaml)
+namespace section to FOO
+- Then run 
+```
+kubectl apply -k manifests/kustomize/env/dev # or other env of your choice
+```
+
+### Disable the public endpoint
+By default, the deployment install an [invert proxy agent](https://github.com/google/inverting-proxy) that exposes a public URL. If you want to skip installing it,
```
Comment (sarahmaddox) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: s/install/installs

Comment (sarahmaddox) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: s/invert/inverting

File: content/docs/pipelines/pipelines-deploy-alternatives.md [hunk 19]
```diff
@@ -0,0 +1,139 @@
...
+- Comment out the proxy component in the [kustomization.yaml](https://github.com/kubeflow/pipelines/blob/master/manifests/kustomize/base/kustomization.yaml).
```
Comment (sarahmaddox) [inline content/docs/pipelines/pipelines-deploy-alternatives.md]: This list does not render properly. See
https://5d8c8937399d410009e3c0a4--competent-brattain-de2d6d.netlify.com/docs/pipelines/pipelines-deploy-alternatives/

**Comment by Bobgy** [comment]
Hi @sarahmaddox, thanks a lot for the detailed review! It's really helpful to me because I'm not a native speaker of English.

I'm currently on vacation, will follow up with fixes after Oct. 7

File: content/docs/pipelines/pipelines-deploy-alternatives.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (Bobgy) [reply content/docs/pipelines/pipelines-deploy-alternatives.md]: @sarahmaddox Thanks for the suggestion.

`alternatives` is because we will soon add documentation for another alternative: GCP marketplace. So I'm just trying to avoid a rename of this piece of document in a short period. What do you think? or maybe we should just rename when we add that?

I will update to `Pipelines Deployment Alternatives` for now and change the file name as you suggested.

File: content/docs/pipelines/pipelines-deploy-alternatives.md (hunk 14 above)
Comment (Bobgy) [reply content/docs/pipelines/pipelines-deploy-alternatives.md]: thanks for the catch! fixed

File: content/docs/pipelines/pipelines-deploy-alternatives.md (hunk 19 above)
Comment (Bobgy) [reply content/docs/pipelines/pipelines-deploy-alternatives.md]: Fixed

**Comment by sarahmaddox** [comment]
Preview: https://deploy-preview-1174--competent-brattain-de2d6d.netlify.com/docs/pipelines/deployment-alternatives/

**Comment by sarahmaddox** [comment]
/lgtm
/approve
/hold

Thanks @Bobgy! This doc LGTM. I've added a hold, to give @IronPan and @jingzhang36 the opportunity to review the page too.

File: content/docs/pipelines/deployment-alternatives.md [hunk 20]
```diff
@@ -0,0 +1,140 @@
++++
+title = "Pipelines Deployment Alternatives"
+description = "Alternative approaches to deploy lightweight Kubeflow Pipelines to a cluster"
+weight = 15
++++
+
+As an alternative to [deploying Kubeflow](/docs/started/getting-started/#installing-kubeflow) as a
```
Comment (jlewi) [inline content/docs/pipelines/deployment-alternatives.md]: Is this GCP only? The instructions below say to use the inverse proxy and I believe that is GCP only?

Comment (Bobgy) [reply content/docs/pipelines/deployment-alternatives.md]: Thanks for the question. I don't know enough context about this.

@IronPan Can you answer after coming back to work?

**Comment by Bobgy** [comment]
@sarahmaddox thanks for the detailed review and approval! @jingzhang36 has already LGTMed. I'm waiting for the team to decide on a consistent branding.

File: content/docs/pipelines/pipelines-deploy-alternatives.md (hunk 4 above)
Comment (Bobgy) [reply content/docs/pipelines/pipelines-deploy-alternatives.md]: The conclusion is "Deploy Kubeflow Pipelines standalone"/"standalone deployment". I will update the documentation.

File: content/docs/pipelines/deployment-alternatives.md (hunk 20 above)
Comment (Bobgy) [reply content/docs/pipelines/deployment-alternatives.md]: @IronPan reminder, do you know the answer?

File: content/docs/pipelines/deployment-alternatives.md [hunk 21]
```diff
@@ -0,0 +1,139 @@
...
+Knowledge about [Kubernetes](https://kubernetes.io/docs/home/), [kubectl](https://kubernetes.io/docs/reference/kubectl/overview/) and [kustomize](https://kustomize.io/) will help you understand this
+document better and be able to customize based on your needs.
+
+
+## Deploying Kubeflow Pipelines standalone to an existing cluster
+
+1. Deploy latest version of Kubeflow Pipelines:
+```
+export PIPELINE_VERSION={{% kfp-latest-version %}}
+kubectl apply -k github.com/kubeflow/pipelines//manifests/kustomize/env/dev?ref=$PIPELINE_VERSION
```
Comment (Bobgy) [inline content/docs/pipelines/deployment-alternatives.md]: I'm consistently using github url to make manifests easier to understand and more discoverable.

**Comment by Bobgy** [comment]
@IronPan reminder, do you want to review this?

Especially, can you take a look at this one https://github.com/kubeflow/website/pull/1174#discussion_r334220179?

**Comment by Bobgy** [comment]
/hold cancel
This is ready to merge after @IronPan reviews

File: content/docs/pipelines/deployment-alternatives.md (hunk 21 above)
Comment (IronPan) [reply content/docs/pipelines/deployment-alternatives.md]: remove the double // in the path
we publish this to gcs now https://github.com/kubeflow/pipelines/blob/master/.release.cloudbuild.yaml#L209

File: content/docs/pipelines/deployment-alternatives.md [hunk 22]
```diff
@@ -0,0 +1,139 @@
++++
+title = "Pipelines Standalone Deployment Alternatives"
```
Comment (IronPan) [inline content/docs/pipelines/deployment-alternatives.md]: is "alternatives" necessary?

File: content/docs/pipelines/deployment-alternatives.md [hunk 23]
```diff
@@ -0,0 +1,139 @@
...
+
+Knowledge about [Kubernetes](https://kubernetes.io/docs/home/), [kubectl](https://kubernetes.io/docs/reference/kubectl/overview/) and [kustomize](https://kustomize.io/) will help you understand this
+document better and be able to customize based on your needs.
+
+
+## Deploying Kubeflow Pipelines standalone to an existing cluster
+
+1. Deploy latest version of Kubeflow Pipelines:
+```
+export PIPELINE_VERSION={{% kfp-latest-version %}}
```
Comment (IronPan) [inline content/docs/pipelines/deployment-alternatives.md]: who will be maintaining kfp-latest-version? 
alternatively we could publish the file to gs://ml-pipeline/pipeline-lite/latest/namespaced-install.yaml
and refer from there. 
wdyt?

File: content/docs/pipelines/deployment-alternatives.md [hunk 24]
```diff
@@ -0,0 +1,139 @@
...
+## Deploying Kubeflow Pipelines standalone from scratch
+
+1. Prepare a Kubernetes cluster:
+
+    You have many options from cloud service providers. For Google Cloud
+    Platform (GCP), you can reference [Creating a cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/creating-a-cluster).
+
+    You can also set up your own Kubernetes cluster.
+
+    Cluster minimum requirement: node pool with at least 2 nodes and 2 CPUs per node.
```
Comment (IronPan) [inline content/docs/pipelines/deployment-alternatives.md]: do you know if this is the minimum requirements for non gcp machines?

File: content/docs/pipelines/deployment-alternatives.md [hunk 25]
```diff
@@ -0,0 +1,139 @@
...
+```
+kubectl describe configmap inverse-proxy-config -n kubeflow | grep googleusercontent.com 
+```
+
+## Deploying Kubeflow Pipelines standalone from scratch
+
+1. Prepare a Kubernetes cluster:
+
+    You have many options from cloud service providers. For Google Cloud
+    Platform (GCP), you can reference [Creating a cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/creating-a-cluster).
```
Comment (IronPan) [inline content/docs/pipelines/deployment-alternatives.md]: inverse proxy is gcp only service. 
can we focus this section on using gcp as an example, including command to create a gke cluster.

File: content/docs/pipelines/deployment-alternatives.md [hunk 26]
```diff
@@ -0,0 +1,139 @@
...
+
+### Change deployment namespace
+To deploy Kubeflow Pipelines standalone in namespace FOO:
+
+- Edit [dev/kustomization.yaml](https://github.com/kubeflow/pipelines/blob/master/manifests/kustomize/env/dev/kustomization.yaml)
+or [gcp/kustomization.yaml](https://github.com/kubeflow/pipelines/blob/master/manifests/kustomize/env/gcp/kustomization.yaml)
+namespace section to FOO.
+- Then run 
+```
+kubectl apply -k manifests/kustomize/env/dev # or other env of your choice
```
Comment (IronPan) [inline content/docs/pipelines/deployment-alternatives.md]: this is confusing to read. maybe just paste the other command. kubectl apply -k manifests/kustomize/env/gcp

File: content/docs/pipelines/deployment-alternatives.md (hunk 20 above)
Comment (IronPan) [reply content/docs/pipelines/deployment-alternatives.md]: yes inverse proxy is gcp only.

Comment (Bobgy) [reply content/docs/pipelines/deployment-alternatives.md]: @IronPan I see, shall we rename this document to deployment alternative for GCP? Will it be possible to be extended to work in other cloud platforms?

File: content/docs/pipelines/deployment-alternatives.md (hunk 22 above)
Comment (Bobgy) [reply content/docs/pipelines/deployment-alternatives.md]: Placeholder for introducing marketplace later

File: content/docs/pipelines/deployment-alternatives.md (hunk 23 above)
Comment (Bobgy) [reply content/docs/pipelines/deployment-alternatives.md]: I think it's important we make the version explicit, it affects what command they use to delete it.

File: content/docs/pipelines/deployment-alternatives.md (hunk 24 above)
Comment (Bobgy) [reply content/docs/pipelines/deployment-alternatives.md]: I don't know, does it work for non gcp since we use inverse-proxy?

File: content/docs/pipelines/deployment-alternatives.md (hunk 25 above)
Comment (Bobgy) [reply content/docs/pipelines/deployment-alternatives.md]: got it, SG

File: content/docs/pipelines/deployment-alternatives.md (hunk 26 above)
Comment (Bobgy) [reply content/docs/pipelines/deployment-alternatives.md]: thanks, will change

File: content/docs/pipelines/deployment-alternatives.md (hunk 23 above)
Comment (sarahmaddox) [reply content/docs/pipelines/deployment-alternatives.md]: +1 to making the version explicit, and to adding the step to the release book. (Doc updates should be part of each release, and part of each feature-development task.)

File: content/docs/pipelines/deployment-alternatives.md (hunk 22 above)
Comment (sarahmaddox) [reply content/docs/pipelines/deployment-alternatives.md]: Thanks for updating the terminology from "lite/lightweight" to "standalone" and for ensuring consistency of this terminology!

Now that we have "standalone" in the title, let's remove "alternatives". We can add it back later if necessary, but it's clearer to keep future possibilities out of the docs until they become a reality. Otherwise people become confused.

File: content/docs/pipelines/deployment-alternatives.md (hunk 20 above)
Comment (Bobgy) [reply content/docs/pipelines/deployment-alternatives.md]: discussed, I will rephrase it to target GCP only

File: content/docs/pipelines/deployment-alternatives.md (hunk 21 above)
Comment (Bobgy) [reply content/docs/pipelines/deployment-alternatives.md]: discussed, we will keep using github url here

File: content/docs/pipelines/deployment-alternatives.md (hunk 22 above)
Comment (Bobgy) [reply content/docs/pipelines/deployment-alternatives.md]: will remove alternatives

File: content/docs/pipelines/deployment-alternatives.md (hunk 23 above)
Comment (Bobgy) [reply content/docs/pipelines/deployment-alternatives.md]: discussed, I will add this to release book

File: content/docs/pipelines/deployment-alternatives.md (hunk 25 above)
Comment (Bobgy) [reply content/docs/pipelines/deployment-alternatives.md]: TODO: make a gcloud example command that works well with all samples

File: content/docs/pipelines/deployment-alternatives.md (hunk 24 above)
Comment (Bobgy) [reply content/docs/pipelines/deployment-alternatives.md]: Tested n1-standard-2 with 2 nodes can run all samples.

File: content/docs/pipelines/deployment-alternatives.md (hunk 25 above)
Comment (Bobgy) [reply content/docs/pipelines/deployment-alternatives.md]: Updated an example

**Comment by Bobgy** [comment]
Hi @sarahmaddox @IronPan, thanks for the suggestions.

I've updated the doc:
* with instructions to create gcp cluster
* remove alternatives
* retarget the doc to gcp only

**Comment by sarahmaddox** [comment]
Latest preview: https://deploy-preview-1174--competent-brattain-de2d6d.netlify.com/docs/pipelines/standalone-deployment-gcp/

File: content/docs/pipelines/standalone-deployment-gcp.md [hunk 27]
```diff
@@ -0,0 +1,178 @@
...
+## Deploying Kubeflow Pipelines standalone to an existing cluster
+
+1. Deploy latest version of Kubeflow Pipelines:
+
+```
+export PIPELINE_VERSION={{% kfp-latest-version %}}
+kubectl apply -k github.com/kubeflow/pipelines//manifests/kustomize/env/dev?ref=$PIPELINE_VERSION
+```
+
+2. Get the URL for the Kubeflow Pipelines UI :
```
Comment (sarahmaddox) [inline content/docs/pipelines/standalone-deployment-gcp.md]: The list formatting is broken - see preview:
https://deploy-preview-1174--competent-brattain-de2d6d.netlify.com/docs/pipelines/standalone-deployment-gcp/

To fix the formatting, indent the code snippets by 4 characters.

File: content/docs/pipelines/standalone-deployment-gcp.md [hunk 28]
```diff
@@ -0,0 +1,178 @@
...
+   Reference:
+
+   - Get gcloud CLI tool at https://cloud.google.com/sdk/gcloud/.
+   - Read `gcloud container clusters create` command documentation at https://cloud.google.com/sdk/gcloud/reference/container/clusters/create.
+
+1. Download kubectl CLI tool:
+
+   You can get kubectl from [its official doc](https://kubernetes.io/docs/tasks/tools/install-kubectl/).
+
+   Requirement: kubectl >=1.14 for native support of kustomize (the `kubectl apply -k` options you will see below).
```
Comment (sarahmaddox) [inline content/docs/pipelines/standalone-deployment-gcp.md]: For clarity, it's best to write this out in full, rather than abbreviate it. So, change this:

```
Requirement: kubectl >=1.14 for ...
```

To this:

```
You need kubectl version 1.14 or later, for ...
```

File: content/docs/pipelines/standalone-deployment-gcp.md [hunk 29]
```diff
@@ -0,0 +1,178 @@
...
+
+1. Configure kubectl to talk to your cluster:
+
+   For GCP, you can reference [Configuring cluster access for kubectl](https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-access-for-kubectl).
```
Comment (sarahmaddox) [inline content/docs/pipelines/standalone-deployment-gcp.md]: Since the doc is now entirely about GCP, we can remove the words "For GCP".

**Comment by sarahmaddox** [comment]
Thanks for these changes @Bobgy! There are just a few formatting things to fix up before we can publish the doc. Once it's published, we can iterate on a bit more polishing. It'll be good to get this doc out there for people to use.

File: content/docs/pipelines/standalone-deployment-gcp.md (hunk 29 above)
Comment (Bobgy) [reply content/docs/pipelines/standalone-deployment-gcp.md]: Thanks, that's right.

File: content/docs/pipelines/standalone-deployment-gcp.md (hunk 28 above)
Comment (Bobgy) [reply content/docs/pipelines/standalone-deployment-gcp.md]: Thanks, your wording does sound better!

File: content/docs/pipelines/standalone-deployment-gcp.md [hunk 30]
```diff
@@ -0,0 +1,178 @@
...
+
+```
+kubectl describe configmap inverse-proxy-config -n kubeflow | grep googleusercontent.com
+```
+
+## Deploying Kubeflow Pipelines standalone from scratch
+
+1. Prepare a Kubernetes cluster:
+
+   You can reference [Creating a cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/creating-a-cluster) for Google Cloud Platform (GCP).
```
[the remark this answers is not in the document]
Comment (Bobgy) [reply content/docs/pipelines/standalone-deployment-gcp.md]: Thanks, I think something went wrong with my autoformatter. Fixed them.

File: content/docs/pipelines/standalone-deployment-gcp.md (hunk 27 above)
[the remark this answers is not in the document]
Comment (Bobgy) [reply content/docs/pipelines/standalone-deployment-gcp.md]: Thanks for the great suggestion! Followed this routine.

Comment (Bobgy) [reply content/docs/pipelines/standalone-deployment-gcp.md]: Fixed

**Comment by Bobgy** [comment]
Thanks for the review @sarahmaddox! I've made suggested changes. Please take a brief review again.

File: content/docs/pipelines/standalone-deployment-gcp.md [hunk 31]
```diff
@@ -22,7 +22,7 @@ export PIPELINE_VERSION={{% kfp-latest-version %}}
 kubectl apply -k github.com/kubeflow/pipelines//manifests/kustomize/env/dev?ref=$PIPELINE_VERSION
 ```
 
-2. Get the URL for the Kubeflow Pipelines UI :
+1. Get the URL for the Kubeflow Pipelines UI :
```
Comment (sarahmaddox) [inline content/docs/pipelines/standalone-deployment-gcp.md]: The list formatting is broken in this section. To fix, indent the code snippets above and below this line.

Comment (Bobgy) [reply content/docs/pipelines/standalone-deployment-gcp.md]: Thanks for the catch!

**Comment by sarahmaddox** [comment]
Thanks!
/lgtm
/approve
